core/src/accounts_hash_verifier.rs

```diff
@@ -479,7 +479,7 @@ impl AccountsHashVerifier {
             hashes.push((accounts_package.slot, *accounts_hash.as_hash()));
         }
 
-        retain_max_n_elements(hashes, MAX_SNAPSHOT_HASHES);
+        retain_max_n_elements(hashes, MAX_LEGACY_SNAPSHOT_HASHES);
```

Comment:
Shouldn't this actually be `MAX_ACCOUNT_HASHES`?

This whole file does not seem to be about snapshot-hashes, but only account-hashes.